src/host_interface/virtio.c

```diff
@@ -252,9 +252,8 @@ static int virtio_process_one(struct virtio_dev* dev, int qidx)
         desc = get_next_desc(q, desc, &idx);
     } while (desc && req->buf_count < VIRTIO_REQ_MAX_BUFS);
 
-    dev->ops->enqueue(dev, qidx, req);
```

Comment:
What's the return value when return is missing? Some random number?

Reply:
Yes, in that case, the return value is undefined (unless the function is `main()`, which returns 0).

Reply:
In theory, it's undefined (and for a while clang would emit `ud2` at the end of any function that returned a value and didn't have a return statement, but people complained).  In practice, it will be whatever was left in the register.  The compiler will likely have tail-called the enqueue function, so the value that it stored in the return register would be seen by the caller of this function, which happened to be what we wanted.

Reply:
Right, but interestingly, it didn't happen each time, which is what led to the bug.